### PR TITLE
[CIS-393] Adjust sorting options

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**
- Adjust `ChannelListSortingKey`
- Adjust `UserListSortingKey`
- Adjust `ChannelMemberListSortingKey`

### CODE REVIEW

File: Sources_v3/Query/Sorting.swift [hunk 1]
```diff
@@ -79,21 +77,20 @@ extension ChannelListSortingKey {
...
+        case .updatedAt: dateKeyPath = \.updatedAt
+        case .lastMessageAt: optionalDateKeyPath = \.lastMessageAt
+        case .memberCount: intKeyPath = \.memberCount
         case .custom: break
         }
         
-        if let keyPath = stringKeyPath {
+        if let keyPath = intKeyPath {
             return .init(keyPath: keyPath, ascending: isAscending)
         }
```
Comment (VojtaStavik) [inline Sources_v3/Query/Sorting.swift:96]: Maybe you can use `PartialKeyPath` for this:
```swift
        var keyPath: PartialKeyPath<ChannelDTO>?
        
        switch self {
        case .default: keyPath = \ChannelDTO.defaultSortingAt
        case .createdAt: keyPath = \ChannelDTO.createdAt
        ...
        }
        
        if let keyPath = keyPath {
            return .init(keyPath: keyPath, ascending: isAscending)
        }
```

File: Sources_v3/Query/Sorting.swift [hunk 2]
```diff
@@ -110,8 +110,9 @@ extension ChannelListSortingKey {
 
 /// `UserListSortingKey` is keys by which you can get sorted users after query.
 public enum UserListSortingKey: SortingKey {
-    case createdAt
-    case updatedAt
+    case id
+    case role
+    case banned
     case lastActiveAt
     case custom(String)
```
Comment (VojtaStavik) [inline Sources_v3/Query/Sorting.swift:117]: Same here, let's add documentation and remove the custom option.

File: Sources_v3/Query/Sorting.swift [hunk 3]
```diff
@@ -170,41 +170,41 @@ extension UserListSortingKey {
 // MARK: - Members Sorting
 
 /// `ChannelMemberListSortingKey` describes the keys by which you can get sorted channel members after query.
-public struct ChannelMemberListSortingKey: RawRepresentable, Equatable, SortingKey {
-    public let rawValue: String
+public enum ChannelMemberListSortingKey: SortingKey {
+    case createdAt
+    case custom(String)
```
Comment (VojtaStavik) [inline Sources_v3/Query/Sorting.swift:175]: ... same here

**Review by VojtaStavik — APPROVED** [review]
Looks OK, thanks! Let's please wait for https://getstream.slack.com/archives/CE5N802GP/p1602573054269000 before merging.

File: Sources_v3/Query/Sorting.swift (hunk 1 above)
Comment (mantergo) [reply Sources_v3/Query/Sorting.swift]: Switched to string key init usage 🙂

**Pull request merged**